File: deviceinfo/src/main/java/com/an/deviceinfo/ads/AdInfo.java
package com.an.deviceinfo.ads;

import android.content.Context;

import java.io.IOException;

public class AdInfo {

    private Context context;
    public AdInfo(Context context) {
        this.context = context;
    }

    public final void getAndroidAdId(final AdIdCallback callback) {
        new Thread(new Runnable() {
            @Override
            public void run() {
                try {

                    AdvertisingIdClient.AdIdInfo adInfo = AdvertisingIdClient.getAdvertisingIdInfo(context);
                    String advertisingId = adInfo.getId();
                    boolean adDoNotTrack = adInfo.isLimitAdTrackingEnabled();

                    //Send Data to callback
                    Ad ad = new Ad();
                    ad.setAdDoNotTrack(adDoNotTrack);
                    ad.setAdvertisingId(advertisingId);
                    callback.onResponse(ad);
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }).start();
    }

    public interface AdIdCallback {
        void onResponse(Ad ad);
    }
}

File: deviceinfo/src/main/java/com/an/deviceinfo/ads/AdInfo.kt
package com.an.deviceinfo.ads

import android.content.Context

import java.io.IOException

class AdInfo(private val context: Context) {

    //Send Data to callback
   val ad: Ad
        @Throws(Exception::class)
        get() {
            val adInfo = AdvertisingIdClient.getAdvertisingIdInfo(context)
            val advertisingId = adInfo.id
            val adDoNotTrack = adInfo.isLimitAdTrackingEnabled
            val ad = Ad(
                advertisingId,
                adDoNotTrack
            )
            return ad
        }

    fun getAndroidAdId(callback: AdIdCallback) {
        Thread(Runnable {
            try {
                val ad = ad
                callback.onResponse(context, ad)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }).start()
    }

    interface AdIdCallback {
        fun onResponse(context: Context, ad: Ad)
    }
}
